src/Sylius/Bundle/ProductBundle/EventListener/SelectProductAttributeChoiceRemoveListener.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ProductBundle\EventListener;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Sylius\Component\Attribute\AttributeType\SelectAttributeType;
+use Sylius\Component\Product\Model\ProductAttributeInterface;
+use Sylius\Component\Product\Model\ProductAttributeValue;
+use Sylius\Component\Product\Model\ProductAttributeValueInterface;
+use Sylius\Component\Product\Repository\ProductAttributeValueRepositoryInterface;
+
+final class SelectProductAttributeChoiceRemoveListener
+{
+    /**
+     * @var string
+     */
+    private $productAttributeValueClass;
+
+    /**
+     * @param string $productAttributeValueClass
+     */
+    public function __construct(string $productAttributeValueClass)
+    {
+        $this->productAttributeValueClass = $productAttributeValueClass;
+    }
+
+    /**
+     * @param LifecycleEventArgs $event
+     */
+    public function postUpdate(LifecycleEventArgs $event): void
+    {
+        /** @var ProductAttributeInterface $productAttribute */
+        $productAttribute = $event->getEntity();
+
+        if (
+            !($productAttribute instanceof ProductAttributeInterface) ||
+            $productAttribute->getType() !== SelectAttributeType::TYPE
```

Comment:
It's my personal preference, but making it a two `if` expressions seems to look cleaner.